### PR TITLE
Arm data

### LINKED ISSUE #368: Incorporate arm data into reference observations
Add arm sites to the `io.reference_obervations`

### PR DESCRIPTION
**Pull request by lboeman**
- [x] Closes #368  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by lboeman** [comment]
Adds a module for incorporating DOE ARM data into the platform. Will require an arm live (https://adc.arm.gov/armlive/) account and api key. I'll try to set up a project specific key for that. Tests are currently failing due to the existence of event data at the dev api.

File: docs/source/whatsnew/1.0.0b5.rst [hunk 1]
```diff
@@ -47,7 +47,10 @@ API Changes
   :py:func:`solarforecastarbiter.metrics.calculator.calculate_probabilistic_metrics` addition. (:issue:`315`) (:issue:`266`) (:pull:`367`)
 * Added api function to get all values of a :py:class:`solarforecastarbiter.datamodel.ProbabilisticForecast` with addition of
   :py:func:`solarforecastarbiter.io.api.get_probabilistic_forecast_values`. (:pull:`367`)
-
+* Added capability to import reference data from DOE ARM sites in the
+  :py:mod:`solarforecastarbiter.io.reference_observations.arm` module. Data can
+  be initialized and updated via the Solar Forecast Arbiter CLI
+  `referencedata` command.(:pull:`390`)
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0b5.rst:53]:
```suggestion
  ``referencedata`` command.(:pull:`390`)
```

File: solarforecastarbiter/io/reference_observations/tests/conftest.py [hunk 2]
```diff
@@ -53,6 +53,17 @@ def site_dicts():
...
                                  "network_api_abbreviation": 'abbrv'},
+        },
+        {
+            'name': 'site4',
+            'latitude': 4,
+            'longitude': -5,
+            'elevation': 12,
+            'timezone': 'Etc/GMT+8',
+            'extra_parameters': {"observation_interval_length": 1,
+                                 "network": 'Unincorportated',
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/tests/conftest.py:64]:
```suggestion
                                 "network": 'Unincorporated',
```

Comment (wholmgren) [reply solarforecastarbiter/io/reference_observations/tests/conftest.py:64]: what is this anyways?

Comment (lboeman) [reply solarforecastarbiter/io/reference_observations/tests/conftest.py:64]: Just ensuring it doesn't try to jam some reference site it doesn't know how to gather data for into the database.

File: solarforecastarbiter/io/reference_observations/sfa_reference_sites.csv [hunk 3]
```diff
@@ -209,3 +209,18 @@ interval_length,name,latitude,longitude,elevation,network_api_id,network_api_abb
...
+1.0,"Southern Great Plains (SGP), Anthony, Kansas",37.1509017944336,-98.36199951171876,417.0,sgpmetE31.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Lamont, Oklahoma",36.60499954223633,-97.48500061035156,307.0,sgpmetE13.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Maple City, Kansas",37.06940078735352,-96.7605972290039,404.0,sgpmetE34.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Marshall, Oklahoma",36.116600036621094,-97.51119995117188,326.0,sgpmetE36.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Medford, Oklahoma",36.81900024414063,-97.81990051269531,335.0,sgpmetE32.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Omega, Oklahoma",35.87969970703125,-98.17279815673828,373.0,sgpmetE38.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Ringwood, Oklahoma",36.430999755859375,-98.28399658203124,414.0,sgpmetE15.b1,sgp,America/Chicago,,DOE ARM
+1.0,"Southern Great Plains (SGP), Tryon, Oklahoma",35.86149978637695,-97.06950378417969,309.0,sgpmetE35.b1,sgp,America/Chicago,,DOE ARM
```
Comment (alorenzo175) [inline solarforecastarbiter/io/reference_observations/sfa_reference_sites.csv:226]: Looks like attribution is hard for the ARM sites, but should we at least link to the pages to generate the proper citation like https://www.arm.gov/capabilities/vaps/qcrad?

Comment (lboeman) [reply solarforecastarbiter/io/reference_observations/sfa_reference_sites.csv:226]: Yeah! I actually hadn't found that citation generator before. Thanks for taking a look for that. I'll add this url to the attribution column of qcrad sites.

**Pull request merged**